cmake/onnxruntime.cmake

```diff
@@ -32,15 +32,24 @@ target_include_directories(onnxruntime PRIVATE ${ONNXRUNTIME_ROOT})
 onnxruntime_add_include_to_target(onnxruntime gsl)
 
 if(UNIX)
-  set(BEGIN_WHOLE_ARCHIVE -Xlinker --whole-archive)
-  set(END_WHOLE_ARCHIVE -Xlinker --no-whole-archive)
-  set(ONNXRUNTIME_SO_LINK_FLAG "-Xlinker --version-script=${SYMBOL_FILE} -Xlinker --no-undefined")
+  if (APPLE)
+    set(BEGIN_WHOLE_ARCHIVE -Xlinker -all_load)
+    set(END_WHOLE_ARCHIVE -Xlinker -noall_load)
+  else()
+    set(BEGIN_WHOLE_ARCHIVE -Xlinker --whole-archive)
+    set(END_WHOLE_ARCHIVE -Xlinker --no-whole-archive)
+    set(ONNXRUNTIME_SO_LINK_FLAG "-Xlinker --version-script=${SYMBOL_FILE} -Xlinker --no-undefined")
+  endif()
 else()
   set(ONNXRUNTIME_SO_LINK_FLAG "-DEF:${SYMBOL_FILE}")
 endif()
 
 if (NOT WIN32)
-  set(CMAKE_SHARED_LINKER_FLAGS "${CMAKE_SHARED_LINKER_FLAGS} -Wl,-rpath='$ORIGIN'")
+  if (APPLE)
+    set_target_properties(onnxruntime PROPERTIES INSTALL_RPATH "@loader_path")
```

Comment:
What is the value of RPATH in the final dylib? Does it get set to $ORIGIN, or @loader_path?

Reply:
@loader_path is the $Origin equivalent for mac